app/Http/Requests/Rdt/RdtApplicantUpdateRequest.php

```diff
@@ -45,4 +47,24 @@ public function rules()
             'symptoms_activity'    => ['sometimes', 'required'],
         ];
     }
+
+    private function ruleNik()
+    {
+        $rule = ['sometimes', 'required'];
+        if ($this->nationality == NationalityEnum::WNI()) {
+            $nik = array_merge($rule, [new NikRule()]); // validation nik
+        } else {
+            $nik = array_merge($rule, ['regex:/^[A-Z0-9]{7,9}$/']); // validation passport, min:7, max:9 character
+        }
```

Comment:
Mang @imamdev93 saya agak concern saja terkait complexity jika ada if else. Bagaimana kalau seperti ini?

```suggestion
        $nikRule = ['regex:/^[A-Z0-9]{7,9}$/']; // validation passport, min:7, max:9 character
        if ($this->nationality == NationalityEnum::WNI()) {
            $nikRule = [new NikRule()]; // validation nik
        }
        $nik = array_merge($rule, $nikRule);
```

Reply:
1. penentuannya dipengkondisiannya
2. merge masing-masing rule array nya di akhir setelah pengkondisian